### PR TITLE
Resolves issue #3528.  Adds support for slices when not using parallel=True.

### PR DESCRIPTION
**Pull request by DrTodd13**
The pyStencil version is still failing.  I think Stuart should take a look at this one.

### CODE REVIEW

File: numba/tests/test_stencils.py [hunk 1]
```diff
@@ -1220,8 +1393,11 @@ def generate_stencil_tree(
         # If astor is installed the decompilation of the AST is also printed
         DEBUG = False
         if DEBUG:
-            print("ORIGINAL")
-            print(ast.dump(tree))
+            try:
+                print("ORIGINAL")
+                print(ast.dump(tree))
+            except ImportError:
```
Comment (sklam) [inline numba/tests/test_stencils.py]: Remove this try-except?

File: numba/tests/test_stencils.py [hunk 2]
```diff
@@ -813,127 +813,300 @@ def visit_Subscript(self, node):
...
+                return newnode
+
+            def computeSlice(i, node):
+                def gen_idx(val, x):
+                    useval = self._const_assigns.get(val, val)
+                    value = self.get_val_from_num(val)
+                    tmp = ast.BinOp(
+                            left=ast.Name(
+                                id=self._id_pat %
+                                self.ids[x],
```
Comment (sklam) [inline numba/tests/test_stencils.py]: This linebreak is odd

File: numba/tests/test_stencils.py [hunk 3]
```diff
@@ -813,127 +813,300 @@ def visit_Subscript(self, node):
...
+                        self._maxes[x] = self._neighborhood[x][1]
+
+                return newnode
+
+            def handle1dindex(node):
+                useval = self._const_assigns.get(
+                    node.slice.value, node.slice.value)
+                idx = ast.BinOp(left=ast.Name(
+                                id=self._id_pat %
+                                self.ids[0],
```
Comment (sklam) [inline numba/tests/test_stencils.py]: Odd linebreak.

File: numba/tests/test_stencils.py [hunk 4]
```diff
@@ -813,127 +813,300 @@ def visit_Subscript(self, node):
...
-                                    ctx=ast.Load()),
+            def handle2dindex(node):
+                idx = []
+                for x, val in enumerate(node.slice.value.elts):
+                    useval = self._const_assigns.get(val, val)
+                    idx.append(
+                        ast.BinOp(
+                            left=ast.Name(
+                                id=self._id_pat %
+                                self.ids[x],
```
Comment (sklam) [inline numba/tests/test_stencils.py]: Odd linebreak.

File: numba/tests/test_stencils.py (hunk 1 above)
Comment (stuartarchibald) [reply numba/tests/test_stencils.py]: Done in https://github.com/numba/numba/pull/3719/commits/5dbca1ff3baac88354934043a83d74442b14b26d

File: numba/tests/test_stencils.py (hunk 3 above)
Comment (stuartarchibald) [reply numba/tests/test_stencils.py]: https://github.com/numba/numba/pull/3719/commits/5dbca1ff3baac88354934043a83d74442b14b26d

File: numba/tests/test_stencils.py (hunk 4 above)
Comment (stuartarchibald) [reply numba/tests/test_stencils.py]: https://github.com/numba/numba/pull/3719/commits/5dbca1ff3baac88354934043a83d74442b14b26d

File: numba/tests/test_stencils.py (hunk 2 above)
Comment (stuartarchibald) [reply numba/tests/test_stencils.py]: https://github.com/numba/numba/pull/3719/commits/5dbca1ff3baac88354934043a83d74442b14b26d

**Comment by stuartarchibald** [comment]
Thanks for the review @sklam, I think https://github.com/numba/numba/pull/3719/commits/5dbca1ff3baac88354934043a83d74442b14b26d addresses all your comments.

**Pull request merged**